nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/CommentManagerImpl.java

```diff
@@ -139,16 +138,29 @@ public CommentManagerImpl(CommentServiceConfig config) {
     public DocumentModel createComment(DocumentModel docModel, String comment, String author) {
         try (CloseableCoreSession session = CoreInstance.openCoreSessionSystem(docModel.getRepositoryName())) {
             DocumentModel commentDM = session.createDocumentModel(COMMENT_DOC_TYPE);
-            commentDM.setPropertyValue(CommentsConstants.COMMENT_TEXT, comment);
-            commentDM.setPropertyValue(CommentsConstants.COMMENT_AUTHOR, author);
-            commentDM.setPropertyValue(CommentsConstants.COMMENT_CREATION_DATE, Calendar.getInstance());
+            commentDM.setPropertyValue(COMMENT_TEXT_PROPERTY, comment);
+            commentDM.setPropertyValue(COMMENT_AUTHOR_PROPERTY, author);
+            commentDM.setPropertyValue(COMMENT_CREATION_DATE_PROPERTY, Calendar.getInstance());
             commentDM = internalCreateComment(session, docModel, commentDM, null);
             session.save();
 
             return commentDM;
         }
     }
 
+    @Override
+    public DocumentModel getThreadForComment(DocumentModel comment) throws CommentSecurityException {
+        List<DocumentModel> threads = getDocumentsForComment(comment);
+        if (threads.size() > 0) {
+            DocumentModel thread = threads.get(0);
+            while (thread.getType().equals("Post") || thread.getType().equals(COMMENT_DOC_TYPE)) {
+                thread = getThreadForComment(thread);
+            }
+            return thread;
+        }
+        return null;
+    }
```

Comment:
I am wondering if we can call directly the `getTopLevelDocument` as if i am not wrong do the same things

Reply:
Probably, as the subject is complex, I would rather go back to the original code where we know its limits. That's why I rollbacked it.